qutip/qip/device/processor.py

```diff
@@ -67,126 +68,275 @@ class Processor(object):
+        Note that they have different requirement for the length of `coeff'.
 
         -"step_func":
         The coefficient will be treated as a step function.
-        E.g. ``tlist=[0,1,2]`` and ``coeffs=[3,2]``, means that the coefficient
+        E.g. ``tlist=[0,1,2]`` and ``coeff=[3,2]``, means that the coefficient
         is 3 in t=[0,1) and 2 in t=[2,3). It requires
-        ``coeffs.shape[1]=len(tlist)-1`` or ``coeffs.shape[1]=len(tlist)``, but
-        in the second case the last element has no effect.
+        ``len(coeff)=len(tlist)-1`` or ``len(coeff)=len(tlist)``, but
+        in the second case the last element of `coeff` has no effect.
 
         -"cubic": Use cubic interpolation for the coefficient. It requires
-        ``coeffs.shape[1]=len(tlist)``
+        ``len(coeff)=len(tlist)``
 
     Attributes
     ----------
     N: int
         The number of component systems.
 
-    ctrls: list
-        A list of the control Hamiltonians driving the evolution.
-
-    tlist: array_like
-        A NumPy array specifies the time of each coefficient.
+    pulses: list of :class:`qutip.qip.Pulse`
+        A list of control pulses of this device
 
-    coeffs: array_like
-        A 2d NumPy array of the shape, the length is dependent on the
-        spline type
-
-    t1: list
-        Characterize the decoherence of amplitude damping for
+    t1: float or list
+        Characterize the decoherence of amplitude damping of
         each qubit.
 
-    t2: list
+    t2: float or list
         Characterize the decoherence of dephasing for
         each qubit.
 
     noise: :class:`qutip.qip.Noise`, optional
         A list of noise objects. They will be processed when creating the
         noisy :class:`qutip.QobjEvo` from the processor or run the simulation.
 
+    drift: :class:`qutip.qip.Drift`
+        A `Drift` object representing the drift Hamiltonians.
+
     dims: list
         The dimension of each component system.
         Default value is a
         qubit system of ``dim=[2,2,2,...,2]``
 
     spline_kind: str
         Type of the coefficient interpolation.
-        "step_func" or "cubic"
+        See parameters of :class:`qutip.qip.Processor` for details.
     """
-    def __init__(self, N, ctrls=None, t1=None, t2=None,
+    def __init__(self, N, t1=None, t2=None,
                  dims=None, spline_kind="step_func"):
         self.N = N
-        self.tlist = None
-        self.coeffs = None
-        self.ctrls = []
-        if ctrls is not None:
-            for H in ctrls:
-                self.add_ctrl(H)
+        self.pulses = []
         self.t1 = t1
         self.t2 = t2
         self.noise = []
+        self.drift = Drift()
         if dims is None:
             self.dims = [2] * N
         else:
             self.dims = dims
         self.spline_kind = spline_kind
 
-    def add_ctrl(self, ctrl, targets=None, cyclic_permutation=False):
+    def add_drift_ham(self, ham, targets, cyclic_permutation=False):
```

Comment:
I would replace instances of `ham` with `hamiltonian`, or remove either `add` or `hamiltonian` to make it shorter

Reply:
Make sense. I'll change it to `add_drift(self, hamiltonian, targets, cyclic_permutation=False)`.